envoy/network/listener.h

```diff
@@ -206,6 +239,11 @@ class ListenerConfig {
    */
   virtual const std::string& name() const PURE;
 
+  /**
+   * @return ListenerInfoConstSharedPtr description of the listener.
+   */
+  virtual const ListenerInfoConstSharedPtr listenerInfo() const PURE;
+
```

Comment:
If there any explicit rule about which field should be part of `ListenerConfig` and which field should be part of `ListenerInfo`? I think all the typedMetadata() and metadata() are part of `ListenerConfig`.

If we only want to expose limited part of the listener to the `FactoryContext`, then I think we can just make the ListenerInfo a base class of `ListenerConfig`.

Reply:
I was thinking about that but not sure if the lifetimes match. Let me look into it.

Reply:
Made it a base class but there's a lifetime problem with QUIC HCM since it latches on to the context but the underlying ListenerImpl is deleted during LDS update. That forces the use of shared_ptr for ListenerInfos in the context implementation.

Reply:
So sorry for the changing. But if there is such a reason, I think it's OK too keep some fields that need additional lifetime in the independent interface `ListenerInfo`.

Reply:
Reverted.